src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1229,6 +1233,15 @@ def edit(
         """Stub."""
         raise NotImplementedError()
 
+    @classmethod
+    def from_result_segments(
+        cls,
+        result_segments: Tuple[BaseSegment, ...],
+        segment_kwargs: Dict[str, Any],
+    ) -> "BaseSegment":
+        """Create a BaseSegment from result segments."""
```

Comment:
I think this docstring can be clearer.

```suggestion
        """Create an instance of this specific segment from a tuple of matched segments."""
```